server/v2/cometbft/server.go

```diff
@@ -68,6 +69,21 @@ func (s *CometBFTServer[T]) Init(appI serverv2.AppI[T], v *viper.Viper, logger l
 		AppTomlConfig:    appTomlConfig,
 	}
 
+	chainID := v.GetString(FlagChainID)
+	if chainID == "" {
+		// fallback to genesis chain-id
+		reader, err := os.Open(filepath.Join(v.GetString(serverv2.FlagHome), "config", "genesis.json"))
```

Comment:
this isnt the safest, nodes dont need the original genesis to run. Im not sure the bet way to get this information on startup? if a db is present we can get it fro there, if its not then we cant safely assume it will be present

Reply:
This is one we have on main/v0.50 right now: https://github.com/cosmos/cosmos-sdk/blob/6b7e9f8935e7ec2af61a3b17c3daab24ab0cfde0/server/util.go#L504-L517

v0.47 got a db check only, as it only happened there (?): https://github.com/cosmos/cosmos-sdk/issues/18781#issuecomment-1872107803

Reply:
hmm, we can keep the same.